sageresearch-app-sdk/src/test/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepositoryTest.kt

```diff
@@ -0,0 +1,397 @@
+
+        verify(activityManager).updateActivities(eq(listOf(scheduledActivity)))
+        testObserver.assertComplete()
+    }
+
+    @Test
+    fun scheduleUpdateFailed_noInternet() {
+        // setup for test
+        val taskStart = mock<Instant>()
+        val taskEnd = mock<Instant>()
+        val taskRunUUID = UUID.randomUUID()
+        val taskResult = mock<TaskResult> {
+            on { taskUUID }.doReturn(taskRunUUID)
+            on { startTime }.doReturn(taskStart)
+            on { endTime }.doReturn(taskEnd)
+        }
+        val scheduleGuid = "scheduleGuid"
+        val scheduledActivityEntity = mock<ScheduledActivityEntity> {
+            on { guid }.doReturn(scheduleGuid)
+        }
+        `when`(scheduledRepositorySyncStateDao.getScheduleGuid(taskRunUUID))
+                .thenReturn(scheduleGuid)
+
+        `when`(scheduledActivityEntityDao.activity(scheduleGuid))
+                .thenReturn(Arrays.asList(scheduledActivityEntity))
+
+        `when`(activityManager.updateActivities(any()))
+                .thenReturn(toV1Single(Single.error<Message>(
+                        Throwable("Unable to resolve host " +
+                                "\"webservices.sagebase.org\", no address associated with hostname"))
+                        .delay(1, SECONDS)))
+
+        // perform action being tested
+        val testObserver = scheduleRepository.updateSchedule(taskResult).test()
+
+        // verify these actions occur after calling updateSchedule but before the response from Bridge
+        inOrder(scheduledRepositorySyncStateDao, scheduledActivityEntity, scheduledActivityEntityDao) {
+            // first to from taskRunUUID -> scheduleGuid
+            verify(scheduledRepositorySyncStateDao).getScheduleGuid(taskRunUUID)
+            // then go from scheduleGuid to entity
+            verify(scheduledActivityEntityDao).activity(scheduleGuid)
+            // we update the schedule using the task's start and end
+            verify(scheduledActivityEntity).startedOn = taskStart
+            verify(scheduledActivityEntity).finishedOn = taskEnd
+            // before doing network call, mark entity as stale and persist our changes to the DB
+            verify(scheduledActivityEntity).needsSyncedToBridge = true
+            verify(scheduledActivityEntityDao).upsert(eq(Arrays.asList(scheduledActivityEntity)))
+        }
+
+        // we should not be done because the Bridge response is still pending
+        testObserver.assertNotComplete()
+
+        // resetting counts/stubs on these
+        reset(scheduledActivityEntityDao, scheduledActivityEntity)
+
+        // ok, now let's let the response come back
+        testSchedulerRule.testScheduler.advanceTimeBy(1, SECONDS)
+
+        // we don't need to change the state on the entity or the DB to retry the sync
+        verifyNoMoreInteractions(scheduledActivityEntity, scheduledActivityEntityDao)
+
+        testObserver.assertComplete()
+    }

```

Comment:
Where did the buildRequestMap tests go?

Reply:
So there's a test file in test and one in androidTest since I didn't migrate all the tests. It should still be there